packages/stable/src/api/geospatial/types.ts

```diff
@@ -0,0 +1,267 @@
+      srid?: number;
+      description?: string;
+      optional?: boolean;
+    }
+  | {
+      type: Exclude<PropertyType, GeometryPropertyType | 'STRING'>;
+      description?: string;
+      optional?: boolean;
+    }
+  | {
+      type: 'STRING';
+      size: number;
+      description?: string;
+      optional?: boolean;
+    };
+
+export interface Properties {
+  [property: string]: Property;
+}
+
+export interface Geospatial {
+  [id: string]: unknown;
+}
+
+export interface FeatureCreateItem extends ExternalId {
+  [property: string]:
+    | string
+    | number
+    | boolean
+    | Date
+    | { wkt: string }
+    | Geometry;
+}
+
+export interface Feature extends FeatureCreateItem {
+  createdTime: Date;
+  lastUpdatedTime: Date;
+}
+
+export interface FeatureType extends ExternalId, InternalId {
+  properties: Properties & {
+    _created_at: { type: 'LONG' };
+    _updated_at: { type: 'LONG' };
+    _external_id: { type: 'STRING'; size: 32 };
+  };
+  searchSpec?: {
+    [indexName: string]: {
+      properties:
+        | Array<keyof Properties>
+        | ['_created_at']
+        | ['_updated_at']
+        | ['_external_id'];
+    };
+  };
+}
+
+export interface FeatureTypeCreateItem extends ExternalId {
+  properties: Properties;
+  searchSpec?: {
+    [indexName: string]: {
+      properties: Array<keyof Properties>;
+    };
+  };
+}
+
+export interface FeatureTypePatch extends ExternalId {
+  update:
+    | {
+        properties: { add: Properties };
+      }
+    | {
+        searchSpec: {
+          add: { [indexName: string]: { properties: Array<keyof Properties> } };
+        };
+      };
+}
+
+export interface FeatureTypeDeleteParams {
+  recursive?: boolean;
+}
+
+export interface FeatureOutputParams {
+  output?:
+    | {
+        geometryFormat: GeometryType;
+      }
+    | {
+        properties: Record<string, unknown>;
+      };
+}
+
+interface AttributeValue {
+  property: string;
+  value: unknown;
+}
+
+interface EqualsFilter {
+  equals: AttributeValue;
+}
+
+interface WithinFilter {
+  within: AttributeValue;
+}
+
+interface IntersectsFilter {
+  intersects: AttributeValue;
+}
+
+interface CompletelyWithinFilter {
+  completelyWithin: AttributeValue;
+}
+
+interface WithinDistanceFilter {
+  withinDistance: AttributeValue & { distance: number };
+}
+
+interface RangeFilter {
+  range:
+    | { property: string; gt: string | number }
+    | { property: string; lt: string | number }
+    | { property: string; gte: string | number }
+    | { property: string; lte: string | number };
+}
+
+interface MissingFilter {
+  missing: { property: string };
+}
+
+interface LikeFilter {
+  like: { property: string; pattern: string };
+}
+
+interface RegexFilter {
+  regex: { property: string; pattern: string | RegExp };
+}
+
+interface OrFilter {
+  or?: NonRecursiveFilters[];
+}
+
+interface AndFilter {
+  and?: NonRecursiveFilters[];
+}
+
+interface NotFilter {
+  not?: NonRecursiveFilters;
+}
+
+interface SortFilter {
```

Comment:
sort is not a filter

Reply:
naming it just Sort seems too generic.
Is SortParams a better name ?

Reply:
We actually call it `orderBy`. Maybe `orderSpec` ?

Reply:
I would recommend using the same name as in the service-contracts repo - if there is a definition already. The consistency might promote ease of use across Cognite, as the type is already named and defined in the documentation.

Ofcourse, it's "strongly" typed, so it's not a big deal. But a nice thought when you're uncertain about naming.

example from the assets api; we see a `ExternalEvent` type referenced.
![image](https://user-images.githubusercontent.com/7851860/147235663-e4c7152c-8764-4a64-8794-bad1fdded6ce.png)

Reply:
Oh that seems like a correct approach @andersfylling 
So according to the following screenshot
<img width="861" alt="Screenshot 2021-12-23 at 1 14 24 PM" src="https://user-images.githubusercontent.com/86822653/147239286-6059fe4e-c9da-446f-b364-881ffe81ebcd.png">

Should I be renaming 
`AllowCrsTransformationFilter` -> `GeospatialAllowCrsTransformation`
`FeatureSearchFilter` -> `GeospatialFeatureFilter`
`FeatureOutputParams` -> `GeospatialOutput`
and so on ?

Reply:
But sort doesn't show any named type, so should I just keep it like string[] ? @andersfylling 
or better to name a `GeospatialSort` or `Sort` to be used ?

Reply:
> Should I be renaming
AllowCrsTransformationFilter -> GeospatialAllowCrsTransformation
FeatureSearchFilter -> GeospatialFeatureFilter
FeatureOutputParams -> GeospatialOutput
and so on ?

Yes. It's up to you, in the end. We do it for everything under the documents service (moving to generating types directly from openapi soon for /documents, so this makes it easier to avoid breaking changes for the grep team).

Reply:
> But sort doesn't show any named type, so should I just keep it like string[] ? @andersfylling
or better to name a GeospatialSort or Sort to be used ?

Does sort need to be a standalone object?

Reply:
Probably sort doesn't need to be standalone. Thanks @andersfylling .